src/artifacts.js

```diff
@@ -4,8 +4,39 @@ let assert  = require('assert');
 let Promise = require('promise');
 let API     = require('taskcluster-lib-api');
 let urllib  = require('url');
+let crypto  = require('crypto');
 let api     = require('./api');
 
+/**
+ * Encode a storage key for storing in the blob storage type backends We do
+ * this to get better partitioning in S3 which gives us better performance.
+ *
+ * Basically `${md5hash(taskId + runId + name).slice(0,7)}/${taskId}/${runId}/${name}`
```

Comment:
Bad docs...
Also do we really need this?
The methods is `let encodeBlobKey = (taskId, runId, name) => `${taskId}/${runId}${name}`;
It seems overkill, it's easier to just have the pattern than to type the function name.